docs/v0.4/getting-started/gke.md

```diff
@@ -0,0 +1,295 @@
+---
+id: gke
+title: Getting started on GKE
+sidebar_label: GKE
+---
+
+The following will help you get started running a riff function with Knative on GKE.
+
+## TL;DR
+
+1. select a Project, install and configure gcloud, kubectl, and helm
+1. create a GKE cluster for Knative
+1. install the latest riff CLI
+1. install Knative using the riff CLI
+1. create a function
+1. invoke the function
+
+
+## create a Google Cloud project
+
+A project is required to consume any Google Cloud services, including GKE clusters. When you log into the [console](https://console.cloud.google.com/) you can select or create a project from the dropdown at the top. 
+
+### install gcloud
+
+Follow the [quickstart instructions](https://cloud.google.com/sdk/docs/quickstarts) to install the [Google Cloud SDK](https://cloud.google.com/sdk/) which includes the `gcloud` CLI. You may need to add the `google-cloud-sdk/bin` directory to your path. Once installed, `gcloud init` will open a browser to start an oauth flow and configure gcloud to use your project.
+
+```sh
+gcloud init
+```
+
+### install kubectl
+[Kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) is the Kubernetes CLI. If you don't already have kubectl on your machine, you can use gcloud to install it.
+
+```sh
+gcloud components install kubectl
+```
+
+### configure gcloud
+
+Create an environment variable, replacing ??? with your project ID (not to be confused with your project name; use `gcloud projects list` to find your project ID). 
+
+```sh
+GCP_PROJECT_ID=???
+```
+
+Check your default project.
+
+```sh
+gcloud config list
+```
+
+If necessary change the default project.
+
+```sh
+gcloud config set project $GCP_PROJECT_ID
+```
+
+List the available compute zones and also regions with quotas.
+
+```sh
+gcloud compute zones list
+gcloud compute regions list
+```
+
+Choose a zone, preferably in a region with higher CPU quota.
+
+```sh
+export GCP_ZONE=us-central1-b
+```
+
+Enable the necessary APIs for gcloud. You also need to [enable billing](https://cloud.google.com/billing/docs/how-to/manage-billing-account) for your new project.
+
+```sh
+gcloud services enable \
+  cloudapis.googleapis.com \
+  container.googleapis.com \
+  containerregistry.googleapis.com
+```
+
+## create a GKE cluster
+
+Choose a new unique lowercase cluster name and create the cluster. For this demo, three nodes should be sufficient.
+
+```sh
+# replace ??? below with your own cluster name
+export CLUSTER_NAME=???
+```
+
+```sh
+gcloud container clusters create $CLUSTER_NAME \
+  --cluster-version=latest \
+  --machine-type=n1-standard-2 \
+  --enable-autoscaling --min-nodes=1 --max-nodes=3 \
+  --enable-autorepair \
+  --scopes=service-control,service-management,compute-rw,storage-ro,cloud-platform,logging-write,monitoring-write,pubsub,datastore \
```

Comment:
I believe the "cloud-platform" scope is a superset that includes most if not all of the others. We need to test and verify of course. FWIW, the Knative-on-GKE install instructions only include "cloud-platform":
https://github.com/knative/docs/blob/master/docs/install/Knative-with-GKE.md